src/main/java/Deadline.java

```diff
@@ -0,0 +1,41 @@
+/**
+ * Deadline class to represent a task with a date to be done by.
+ */
+
+import java.text.ParseException;
+import java.util.Date;
+import java.text.SimpleDateFormat;
+
+public class Deadline extends Task {
+
+    protected String by;
+    protected Date date;
+
+    public Deadline(String description, String by) throws ParseException {
+        super(description);
+        this.by = by;
```

Comment:
Just wondering, why do you need by and date attributes? As I was thinking that since they are similar representations of the same thing, and date can also be used in the toString() method, it is alright to keep date only. :)